airflow/pipe_naf_reader_dag.py

```diff
@@ -0,0 +1,127 @@
+            delegate_to=self.delegate_to)
+        return len(hook.list(self.bucket, prefix=self.prefix)) > 0
+
+
+def source_exists_sensor(bucket, prefix, dag):
+    return GoogleCloudStoragePrefixSensor(
+        dag=dag,
+        task_id='source_exists',
+        bucket=bucket,
+        prefix=prefix,
+        poke_interval=10,   # check every 10 seconds for a minute
+        timeout=60,
+        retries=24*7,       # retry once per hour for a week
+        retry_delay=timedelta(minutes=60),
+        retry_exponential_backoff=False
+    )
+
+
+class NAFReaderDagFactory(DagFactory):
+
+    def __init__(self, pipeline=PIPELINE, **kwargs):
+        super(NAFReaderDagFactory, self).__init__(pipeline=pipeline, **kwargs)
+
+    def source_date(self):
+        if self.schedule_interval == '@daily':
+            return '{{ ds }}','{{ ds_nodash }}'
+        else:
+            raise ValueError('Unsupported schedule interval {}'.format(self.schedule_interval))
+
+    def build(self, dag_id):
+        config = self.config
+        source_date = self.source_date()
+        config['date'] = source_date[0]
+        config['date_nodash'] = source_date[1]
+        print('config = %s' % (config))
+        country_configurations = json.loads(config['configurations'])
+
+
+        naf_reader_image = "{docker_image}".format(**config)
+        print('naf_reader_image = %s' % naf_reader_image)
+
+        dags=[]
+        for country_config in country_configurations:
+            dag_id='{}.{}'.format(dag_id)
+            naf_reader_args = ['{name}',
+                               '{gcs_source}',
+                               '{gcs_csv_output}',
+                               '{bq_output}',
+                               '{ds}']
+            for arg in range(len(naf_reader_args)):
+                naf_reader_args[arg]=naf_reader_args[arg].format(**config)
+            naf_reader_args.insert(0, 'naf_reader_daily')
+            print('naf_reader_args = %s' % naf_reader_args)
+            naf_reader_args.insert(0, "./scripts/run.sh")
+
+            with DAG(dag_id, schedule_interval=self.schedule_interval, default_args=self.default_args) as dag:
+
+                # Checks that the gcs folder exists and contains file for the current {ds}
+                # Question? must be inside the k8s op
+                source_bucket = re.search('(?<=gs://)[^/]*', country_config['gcs_source']).group(0)
+                naf_gcs_path = re.search('(?<=gs://)[^/]*/(.*)', country_config['gcs_source']).group(1)
+                naf_gcs_path = '{}/{}'.format(naf_gcs_path, config['ds'])
+                source_exists = source_exists_sensor(
+                    bucket=source_bucket,
+                    prefix=naf_gcs_path,
+                    dag=dag
+                )
+
+                naf_reader = KubernetesPodOperator(
+                    namespace=os.getenv('K8_NAMESPACE'),
+                    image=naf_reader_image,
+                    cmds=naf_reader_args,
```

Comment:
I would pass parameter explicitely here, doing a `for` loop don't brings any benefit and it makes code reading/mantainance a bit harder:

```
cmds='scripts/run.sh  '
                             '{source_dates} '
                             '{source_paths} '
                             '{project_id}:{pipeline_dataset}.{normalized} '
                             ''.format(**config)
```